google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplTest.java

```diff
@@ -3072,4 +3078,183 @@ public void testWriterWithSignedURL() throws MalformedURLException {
     assertNotNull(writer);
     assertTrue(writer.isOpen());
   }
+
+  @Test
+  public void testUploadNonExistentFile() {
+    EasyMock.replay(storageRpcMock);
+    initializeService();
+    String fileName = "non_existing_file.txt";
+    try {
+      storage.upload(BLOB_INFO1, Paths.get(fileName));
+      Assert.fail();
+    } catch (IOException e) {
+      assertEquals(NoSuchFileException.class, e.getClass());
+      assertEquals(fileName, e.getMessage());
+    }
+  }
+
+  @Test
+  public void testUploadDirectory() throws IOException {
+    EasyMock.replay(storageRpcMock);
```

Comment:
Please use Mockito for all new tests. It really is better.

Reply:
@elharo I was going to use Mockito for these new tests, but found this idea not very productive.
I submitted #270 and assigned to myself to fix this.

Reply:
@elharo, new tests for upload() converted to mockito.
It's not a final version yet. upload() method will be updated to return Blob (now they return void), that will require test update as well.